lib/improver/tests/calculate_sleet_prob/test_calculate_sleet_probability.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the calculate_sleet_probability plugin."""
+
+import unittest
+
+import numpy as np
+from iris.tests import IrisTest
+from improver.calculate_sleet_prob import calculate_sleet_probability
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+
+
+class Test_calculate_sleet_probability(IrisTest):
+    """ Tests the calculate sleet probability function."""
+
+    def setUp(self):
+        """Create cubes to input into the function."""
+
+        rain_prob = np.array([[0.5, 0.1, 1.0],
+                              [0.0, 0.2, 0.5],
+                              [0.1, 0.1, 0.3]], dtype=np.float32)
+        self.rain_prob_cube = set_up_variable_cube(rain_prob)
+
+        snow_prob = np.array([[0.0, 0.4, 0.0],
+                              [0.5, 0.3, 0.1],
+                              [0.0, 0.4, 0.3]], dtype=np.float32)
+        self.snow_prob_cube = set_up_variable_cube(snow_prob)
+
```

Comment:
If we can't rule out cases with P(snow) + P(rain) > 1 we need a test to cover this and ensure that the plugin does not return negative probabilities.

Reply:
I'll add a test to the plugin for this.

Reply:
We could do with some testing of the metadata in the output cubes.

I presume all the input cubes here are actually just air temperature as the set_up_variable_cube function is being invoked without any modifications. There is a specific function for setting up probability cubes in the same file which would be more appropriate here I guess (assuming it can handle a single threshold).

Reply:
I have added a test to check that the name of the cube changes to probability_of_sleet through the plugin. This name may change but the tests/plugin can be updated if necessary.